src/astro/files/__init__.py

```diff
@@ -14,3 +16,26 @@ def check_if_connection_exists(conn_id: str) -> bool:
     """
     BaseHook.get_connection(conn_id)
     return True
+
+
+def get_file_list(path: str, conn_id: str, **kwargs) -> TaskDecorator:
```

Comment:
Have you considered using:
```get_file_list(files_prefix:File)```

Since `File` already abstracts both properties:
```File(path="s3://some-bucket/prefix/*", conn_id="some_conn")```

Reply:
hmm, I think it would make the public interface more consistent. but getting error when creating below object 
```
Broken DAG: [/usr/local/airflow/dags/task_flow.py] Traceback (most recent call last):
  File "/usr/local/airflow/astro_sdk/src/astro/files/types/__init__.py", line 25, in create_file_type
    filetype = get_filetype(path)
  File "/usr/local/airflow/astro_sdk/src/astro/files/types/__init__.py", line 53, in get_filetype
    raise ValueError(
ValueError: Missing file extension, cannot automatically determine filetype from path 'gs://pankaj_test_dtt/'. Please pass the 'filetype' param with the explicit filetype (e.g. csv, ndjson, etc.).
```
here I want to list files from ``pankaj_test_dtt``

Reply:
hmm thinking of this more, this would be a problem from load_file too I think i.e it might fail if we pass `File(path="gs://pankaj_test_dtt/folder1/*", conn_id="bigquery_conn_id")`

cc @utkarsharma2 @sunank200 ^^ -- Can we test this before 1.0 please

Reply:
The problem was because the list returned by `FileLocation.path` contained "folders" (they are not folder per see on Cloud Storage) too, example:

```
In [13]: [urlunparse((url.scheme, url.netloc, keys, "", "", "")) for keys in prefixes]
Out[13]:
['gs://dag-authoring/dynamic_task/',
 'gs://dag-authoring/dynamic_task/homes.csv']
```

This is why the operator failed to determine filetype for first path.

Now, this is valid for Native support, but specifically not needed for `get_file_list`, `ListFileOperator` so I have pushed https://github.com/astronomer/astro-sdk/pull/596/commits/adf94c5163bff151ecca58cae6f022566708df5c for this PR. But we should still think about `load_file` operator, if somehow the "files" are loaded twice, as path 1 'gs://dag-authoring/dynamic_task/' is superset of 'gs://dag-authoring/dynamic_task/homes.csv'.

Can one of you check this tomorrow please?

Reply:
got it. So @kaxil does it mean currently we can't change

```def get_file_list(path: str, conn_id: str, **kwargs)``` ->  ```def get_file_list(file: File, **kwargs)```

here file, can be a bucket folder path?

Reply:
discussed with @utkarsharma2 currently, if want user to pass file object then they will have to pass file type too like below 
```
get_file_list(File(path=GCS_BUCKET, conn_id=ASTRO_GCP_CONN_ID, filetype=FileType.CSV))
```

Reply:
created an issue to fix pattern searching https://github.com/astronomer/astro-sdk/issues/642

Reply:
@tatiana for now we will go wiouthout file object because would restrict us and user will have to pass filetype also

I have created a ticket to make file object more flexible https://github.com/astronomer/astro-sdk/issues/644